Source/WebCore/platform/graphics/texmap/coordinated/GraphicsLayerCoordinated.cpp

```diff
@@ -79,19 +79,27 @@ std::optional<PlatformLayerIdentifier> GraphicsLayerCoordinated::primaryLayerID(
 
 void GraphicsLayerCoordinated::setNeedsDisplay()
 {
-    if (!m_drawsContent || !m_contentsVisible || m_size.isEmpty() || m_dirtyRegion.fullRepaint)
+    if (!m_drawsContent || !m_contentsVisible || m_size.isEmpty())
         return;
 
-    m_dirtyRegion.fullRepaint = true;
-    m_dirtyRegion.rects.clear();
-    noteLayerPropertyChanged(Change::DirtyRegion, ScheduleFlush::Yes);
+    if (m_dirtyRegion) {
+        if (m_dirtyRegion->mode() == Damage::Mode::Full)
+            return;
+
+        m_dirtyRegion->makeFull(m_size);
```

Comment:
Would it make sense to move the `mode() == Damange::Mode::Full` check into the makeFull method?

Reply:
It's already there, but in this particular case I need to return early if it's already full.